### DISCUSSION TITLE
Incremental output not working with "webpack --mode=development --watch"

### DISCUSSION THREAD

**Discussion by CalculonPrime** [webpack/webpack#15661] [General]
I have a project with multiple html/jsx pages under a central directory.  When I run "webpack --mode=development --watch" I was expecting that only the chunks generated from changed jsx would be re-emitted into the build directory.  Instead, chunks are re-emitted for ALL of the pages under the central directory when the jsx on any one page is modified.

I don't know necessarily that unchanged chunks are being recompiled, but they are being re-emitted, with different mangling extensions (e.g. foo.61e8.js is added to a subdirectory that before only had foo.ea5d.js)  I wouldn't expect _any_ files to be re-emitted that haven't changed.

Should I file an Issue for this to get it fixed?

I'm using Webpack 5.70.0

**Comment by vankop** [comment]
~~what you mean "reemitted"? In console log they should be marked as `[emitted]`, but they probably were not write to fs.~~

Could you provide repro?

**Reply by vankop** [reply]
> with different mangling extensions (e.g. foo.61e8.js is added to a subdirectory that before only had foo.ea5d.js)

maybe you have `output.chunkFilename` with `chunkhash` ( chunk hash could be changed ). try to use `[contenthash]`
